snuba/migrations/groups.py

```diff
@@ -41,323 +52,39 @@ class MigrationGroup(Enum):
-    def __init__(self) -> None:
-        super().__init__("snuba.snuba_migrations.test_migration")
-
-    def get_migrations(self) -> Sequence[str]:
-        return ["0001_create_test_table", "0002_add_test_col"]
-
-
-class ProfilesLoader(DirectoryLoader):
-    def __init__(self) -> None:
-        super().__init__("snuba.snuba_migrations.profiles")
-
-    def get_migrations(self) -> Sequence[str]:
-        return [
-            "0001_profiles",
-            "0002_disable_vertical_merge_algorithm",
-            "0003_add_device_architecture",
-            "0004_drop_profile_column",
-        ]
-
-
-class FunctionsLoader(DirectoryLoader):
-    def __init__(self) -> None:
-        super().__init__("snuba.snuba_migrations.functions")
-
-    def get_migrations(self) -> Sequence[str]:
-        return ["0001_functions"]
-
-
-class GenericMetricsLoader(DirectoryLoader):
-    def __init__(self) -> None:
-        super().__init__("snuba.snuba_migrations.generic_metrics")
-
-    def get_migrations(self) -> Sequence[str]:
-        return [
-            "0001_sets_aggregate_table",
-            "0002_sets_raw_table",
-            "0003_sets_mv",
-            "0004_sets_raw_add_granularities",
-            "0005_sets_replace_mv",
-            "0006_sets_raw_add_granularities_dist_table",
-            "0007_distributions_aggregate_table",
-            "0008_distributions_raw_table",
-            "0009_distributions_mv",
-            "0010_counters_aggregate_table",
-            "0011_counters_raw_table",
-            "0012_counters_mv",
-        ]
-
-
-class SearchIssuesLoader(DirectoryLoader):
-    def __init__(self) -> None:
-        super().__init__("snuba.snuba_migrations.search_issues")
-
-    def get_migrations(self) -> Sequence[str]:
-        return [
-            "0001_search_issues",
-            "0002_search_issues_add_tags_hash_map",
-            "0003_search_issues_modify_occurrence_type_id_size",
-            "0004_rebuild_search_issues_with_version",
-            "0005_search_issues_v2",
-        ]
-
-
-_REGISTERED_GROUPS = {
-    MigrationGroup.SYSTEM: SystemLoader(),
-    MigrationGroup.EVENTS: EventsLoader(),
-    MigrationGroup.TRANSACTIONS: TransactionsLoader(),
-    MigrationGroup.DISCOVER: DiscoverLoader(),
-    MigrationGroup.METRICS: MetricsLoader(),
-    MigrationGroup.OUTCOMES: OutcomesLoader(),
-    MigrationGroup.SESSIONS: SessionsLoader(),
-    MigrationGroup.QUERYLOG: QuerylogLoader(),
-    MigrationGroup.PROFILES: ProfilesLoader(),
-    MigrationGroup.FUNCTIONS: FunctionsLoader(),
-    MigrationGroup.REPLAYS: ReplaysLoader(),
-    MigrationGroup.GENERIC_METRICS: GenericMetricsLoader(),
-    MigrationGroup.TEST_MIGRATION: TestMigrationLoader(),
-    MigrationGroup.SEARCH_ISSUES: SearchIssuesLoader(),
+class _MigrationGroup:
+    def __init__(self, loader: GroupLoader, readiness_state: ReadinessState) -> None:
+        self.loader = loader
+        self.readiness_state = readiness_state
+
+
+_REGISTERED_MIGRATION_GROUPS = {
```

Comment:
yeah this seems ripe for RegisteredClass, but I see now that you're just moving these files around, not changing them?

Reply:
Yes, one of the reason for decoupling the loaders and MigrationGroups files is to be able to build guardrails to check that migration changes (include `readiness_state` changes) are not deployed alongside code changes.